nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -306,6 +319,7 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
     RCLCPP_WARN(logger_, "Unable to retrieve goal checker's tolerances!");
   } else {
     goal_dist_tol_ = pose_tolerance.position.x;
+    goal_yaw_tol_ = tf2::getYaw(pose_tolerance.orientation);
```

Comment:
Hmm the same is true for `goal_dist_tol_` above? I'm just adding missing information
Goal is always going to be in either `map` or `odom` for the navigation stack.. i'm not sure anything would work otherwise?

Reply:
Even if the goal is in the map frame, axis may point other directions. Take our fiducial mapping for example, poses are in map frame, but Z is forward

Reply:
When does map/odom frame have its z axis pointed down?

Reply:
Map/odom has predefined axis direction, but the goal in map/odom frame might have any direction of axis, including z down

Reply:
Hmm.. How would I handle that?
Also this line of code grabs the goal **tolerence** from goal checker

Reply:
This for example
https://eigen.tuxfamily.org/dox/classEigen_1_1QuaternionBase.html#title10

Reply:
Discussed with Dmitry - To be able to handle arbitrary 3D goal pose goes beyond the scope of this PR, as it may require changes in controller_server, goal_checkers, planner_server, and planners and possibly other places